tfx/components/example_gen/write_split.py

```diff
@@ -48,22 +52,36 @@ def process(self, e: Union[tf.train.Example, tf.train.SequenceExample,
 
 @beam.ptransform_fn
 @beam.typehints.with_input_types(
-    Union[tf.train.Example, tf.train.SequenceExample, bytes])
+    Union[tf.train.Example, tf.train.SequenceExample, bytes, pa.Table])
 @beam.typehints.with_output_types(beam.pvalue.PDone)
 def WriteSplit(
     example_split: beam.pvalue.PCollection,
     output_split_path: str,
     output_format: str,
+    exec_properties: Dict[str, Any]
 ) -> beam.pvalue.PDone:
-  """Shuffles and writes output split as serialized records in TFRecord."""
+  """Shuffles and writes output split as serialized records in TFRecord or Parquet."""
   del output_format
 
+  output_payload_format = exec_properties.get(
+    standard_component_specs.OUTPUT_DATA_FORMAT_KEY)
+
+  if output_payload_format == example_gen_pb2.PayloadFormat.FORMAT_PARQUET:
+    # TODO: propagate schema.
+    schema = exec_properties.get("pyarrow_schema")
+    return (example_split
+            # TODO(jyzhao): make shuffle optional.
+            | 'Shuffle' >> beam.transforms.Reshuffle()
+            | "Write Parquet" >> beam.io.WriteToParquet(
+              os.path.join(output_split_path, DEFAULT_PARQUET_FILE_NAME),
+              schema, file_name_suffix=".parquet", codec='snappy'))
```

Comment:
just confirm, this can be read by downstream tfxio, right?

and question for Ihor, is there a tfxio writing util function that ready to be used?

Reply:
Working on this atm, so not yet.

it's going to look something like:
```
(record_batches | 'RecordBatchToDicts'  >> beam.FlatMap(lambda x: x.to_pandas().to_dict(orient='records'))
| 'WriteParqet' >> beam.io.WriteToParquet(output_path, schema, file_name_suffix=".parquet", codec='snappy'))
```

the arrow schema will be inferred from the transformed tfmd schema